Orckestra.Widget.FilteredSelector/Orckestra.Widget.FilteredSelector/WidgetProvider/FilteredSelectorWidgetProcessingManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+using Composite.Core.Types;
+using Composite.Core.WebClient.Renderings.Page;
+using Composite.Data;
+
+namespace Orckestra.Widget.FilteredSelector.WidgetProvider
+{
+    internal static class FilteredSelectorWidgetProcessingManager
+    {
+        public static IEnumerable<Guid> GetAssociatedPageIds(Guid pageId, SitemapScope sitemapScope) => PageStructureInfo.GetAssociatedPageIds(pageId, sitemapScope);
+        private static IEnumerable<KeyValuePair> GetOptionsForDefault(Type t, Guid pageId, SitemapScope sitemapScope)
+        {
+            IEnumerable<Guid> availablePageIds = GetAssociatedPageIds(pageId, sitemapScope).Distinct();
+
+            using (DataConnection con = new DataConnection())
+            {
+                MethodInfo generic = typeof(DataConnection).GetMethod("Get").MakeGenericMethod(t);
+                IEnumerable<KeyValuePair> result = from a in (IEnumerable<IPageRelatedData>)generic.Invoke(con, null)
+                                                   join b in availablePageIds
+                                                   on a.PageId equals b
+                                                   select new KeyValuePair(a.GetKeyProperties().First(k => k.Name == "Id").GetValue(a).ToString(), a.GetLabel());
+                return result;
+            }
+        }
+
+        internal static IEnumerable GetParameters(string parameters)
+        {
+            Match match = Regex.Match(parameters, 
+                $"{Constants.pageIdParamName}:\"(.+?)\";{Constants.typeNameParamName}:\"(.+?)\";{Constants.sitemapScopeIdParamName}:\"([0-9]+)\"");
+
+            if (match.Success == false)

```

Comment:
Please use :) 

`!match.Success`

Reply:
Updated, will keep this style in this project :)